### PR TITLE
[TASK] migrate plugin subtype "list_type"

### PR DESCRIPTION
**Pull request by dkd-kaehm**
Migrates Plugin subtypes to content elements.
See:  https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/13.4/Deprecation-105076-PluginContentElementAndPluginSubTypes.html#deprecation-105076-plugin-content-element-and-plugin-sub-types

Fixes: #4180
Relates: #3995

---

### Maintainers notes:

### CODE REVIEW

File: Classes/Updates/SolrSearchCTypeMigration.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+namespace ApacheSolrForTypo3\Solr\Updates;
+
+use TYPO3\CMS\Install\Attribute\UpgradeWizard;
+use TYPO3\CMS\Install\Updates\AbstractListTypeToCTypeUpdate;
+
+/**
+ * Migrates Plugin subtypes to content elements.
+ * See: https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/13.4/Deprecation-105076-PluginContentElementAndPluginSubTypes.html#deprecation-105076-plugin-content-element-and-plugin-sub-types
+ */
+#[UpgradeWizard('solrSearchCTypeMigration')]
```
Comment (dkd-friedrich) [inline Classes/Updates/SolrSearchCTypeMigration.php:27]: We need to enable `autoconfigure` and `autowire` for `ApacheSolrForTypo3\Solr\Updates` to ensure the wizard can be found:

```
  ApacheSolrForTypo3\Solr\Updates\:
    resource: '../Classes/Updates/*'
    autoconfigure: true
    autowire: true
```

File: Classes/Updates/SolrSearchCTypeMigration.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    public function getTitle(): string
+    {
+        return 'Migrate "Apache Solr Search" plugins to content elements.';
+    }
+
+    public function getDescription(): string
+    {
+        return 'The "Apache Solr Search" plugins are now registered as content element. Update migrates existing records and backend user permissions.';
```
Comment (dkd-friedrich) [inline Classes/Updates/SolrSearchCTypeMigration.php:46]: I think a `s` is missing and it should be `The "Apache Solr Search" plugins are now registered as content elements.`.

File: Configuration/TCA/Overrides/tt_content.php [hunk 3]
```diff
@@ -5,44 +5,43 @@
 
 defined('TYPO3') or die('Access denied.');
 
+// Add tt_content search group
+$GLOBALS['TCA']['tt_content']['columns']['CType']['config']['itemGroups']['search'] = 'LLL:EXT:solr/Resources/Private/Language/locallang.xlf:plugin_results';
```
Comment (dkd-friedrich) [inline Configuration/TCA/Overrides/tt_content.php:9]: Should we use a more unique key like `tx_solr` for the item group?

**Review by dkd-friedrich — CHANGES_REQUESTED** [review]

File: Configuration/TCA/Overrides/tt_content.php [hunk 4]
```diff
@@ -5,44 +5,43 @@
...
     'solr',
     'pi_frequentlySearched',
-    'LLL:EXT:solr/Resources/Private/Language/locallang.xlf:tt_content.list_type_pi_frequentsearches'
+    'LLL:EXT:solr/Resources/Private/Language/locallang.xlf:tt_content.CType_pi_frequentsearches',
+    'extensions-solr-plugin-contentelement',
+    'search'
 );
-$GLOBALS['TCA']['tt_content']['types']['list']['subtypes_excludelist'][$pluginSignature]
-    = 'layout,select_key,pages,recursive';
+$GLOBALS['TCA']['tt_content']['types'][$pluginFrequentlySearchedSignature]['showitem'] = 'pi_flexform';
```
Comment (dkd-friedrich) [inline Configuration/TCA/Overrides/tt_content.php:33]: There is no flexform for frequent search plugin

File: Configuration/TCA/Overrides/tt_content.php (hunk 3 above)
Comment (dkd-kaehm) [reply Configuration/TCA/Overrides/tt_content.php:9]: Internally discussed, we'll use "search".

**Pull request merged**